Mathlib/Analysis/Analytic/IsolatedZeros.lean

```diff
@@ -217,6 +217,27 @@ lemma order_eq_nat_iff (hf : AnalyticAt 𝕜 f z₀) (n : ℕ) : hf.order = ↑n
     refine ⟨fun hn ↦ (WithTop.coe_inj.mp hn : h.choose = n) ▸ h.choose_spec, fun h' ↦ ?_⟩
     rw [unique_eventuallyEq_pow_smul_nonzero h.choose_spec h']
 
+/- An analytic function `f` has finite order at a point `z₀` iff locally looks
+  like `(z - z₀) ^ order • g`, where `g` is analytic and does not vanish at
+  `z₀`. -/
+lemma order_neq_top_iff (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order ≠ ⊤ ↔ ∃ (g : 𝕜 → E), AnalyticAt 𝕜 g z₀ ∧ g z₀ ≠ 0
+      ∧ f =ᶠ[𝓝 z₀] fun z ↦ (z - z₀) ^ (hf.order.toNat) • g z := by
+  erw [← hf.order_eq_nat_iff]
+  exact ⟨fun h₁f ↦ (ENat.coe_toNat h₁f).symm, fun h₁f ↦ ENat.coe_toNat_eq_self.mp h₁f.symm ⟩
+
+/- An analytic function has order zero at a point iff it does not vanish there. -/
+lemma order_eq_zero_iff (hf : AnalyticAt 𝕜 f z₀) :
+    hf.order = 0 ↔ f z₀ ≠ 0 := by
+  rw [(by rfl : (0 : ENat) = (0 : Nat)), order_eq_nat_iff hf 0]
+  constructor
+  · intro ⟨g, _, _, hg⟩; simpa [Eventually.self_of_nhds hg]
+  · intro hz; use f; exact ⟨hf, hz, by simp⟩
```

Comment:
You can definitely compress this line, by making `f` part of the exact.

You can also write it as `exact fun hz \mapsto  ⟨f, hf, hz, by simp⟩` (similarly for the line above);
then you could combine this into one big
```
exact ⟨fun  ⟨g, _, _, hg⟩ \mapsto by simpa [Eventually.self_of_nhds hg],
  fun hz \mapsto ⟨f, hf, hz, by simp⟩
```
with \mapsto replaced by the corresponding symbol.

I think the first suggestion is definitely better. This second one might be a matter of taste. (If you think that "this proof is mathematically trivial", I'd say you can golf it down all the way. If not, using constructor is fine.)

Reply:
Implemented following your suggestion #1.